src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java

```diff
@@ -60,44 +77,103 @@ public void run() {
         });
         readerThread.start();
 
-        int readerPort= getPortNumber();
+        int readerPort = getPortNumber();
         try {
-            String bindHost = getBindHost();
+            Security.addProvider(new BouncyCastleProvider());
             HttpServer server = null;
-            if (bindHost != null && !bindHost.trim().isEmpty()) {
-                LOG.info("Binding to Interface: {}", bindHost);
-                server = HttpServer.create(new InetSocketAddress(InetAddress.getByName(bindHost.trim()), readerPort),
-                                           INCOMING_QUEUE_LENGTH);
-            } else {
-                LOG.info("Value Not Configured for: {} Using default value: binding to all interfaces", WEBSERVICE_BIND_HOST_NAME);
-                server = HttpServer.create(new InetSocketAddress(readerPort), INCOMING_QUEUE_LENGTH);
+            if(PluginSettings.instance().getHttpsEnabled()) {
+                server = createHttpsServer(readerPort);
+            }
+            else {
+                server = createHttpServer(readerPort);
             }
             server.createContext(QUERY_URL, new QueryMetricsRequestHandler());
-
             server.setExecutor(Executors.newCachedThreadPool());
             server.start();
-        } catch(java.net.BindException ex) {
+        } catch (java.net.BindException ex) {
             LOG.error("Port  {} is already in use...exiting", readerPort);
             Runtime.getRuntime().halt(1);
-        } catch(Exception ex) {
+        } catch (Exception ex) {
             LOG.error("Exception in starting Reader Process: " + ex.toString());
             Runtime.getRuntime().halt(1);
         }
     }
 
+    private static HttpServer createHttpsServer(int readerPort) throws Exception {
```

Comment:
should the return type be HttpsServer?

Reply:
HttpsServer is a subclass of HttpServer.